src/test/unit/lambda_tracer/test_lambda_reporter.py

```diff
@@ -73,7 +228,67 @@ def test_create_request_body_take_error_first(dummy_span, error_span, function_e
         function_end_span,
     ]
     size = _get_event_base64_size(expected_result)
-    assert _create_request_body(input, True, size) == json.dumps(expected_result)
+    assert _create_request_body(
+        input_spans, True, max_size=size, max_error_size=size
+    ) == json.dumps(expected_result)
+
+
+def test_create_request_body_take_only_metadata_function_span(
+    function_end_span: dict, function_end_span_metadata: dict
+):
+    expected_result = [function_end_span_metadata]
+    input_spans = [function_end_span]
+    size = _get_event_base64_size(expected_result)
+
+    result = _create_request_body(input_spans, True, max_size=size, max_error_size=size)
+
+    assert result == json.dumps(expected_result)
+
+
+def assert_use_metadata_span_when_needed(
+    function_span, wrapper_span, wrapper_span_metadata
+) -> None:
+    expected_result = [function_span, wrapper_span_metadata]
+    input_spans = [wrapper_span, function_span]
+    size = _get_event_base64_size(expected_result)
+
+    result = _create_request_body(input_spans, True, max_size=size, max_error_size=size)
+
+    assert result == json.dumps(expected_result)
+
+
+def test_create_request_body_take_only_metadata_redis_span(
+    function_end_span: dict, redis_span: dict, redis_span_metadata: dict
+):
+    assert_use_metadata_span_when_needed(function_end_span, redis_span, redis_span_metadata)
+
+
+def test_create_request_body_take_only_metadata_http_span(
+    function_end_span: dict, http_span: dict, http_span_metadata: dict
+):
+    assert_use_metadata_span_when_needed(function_end_span, http_span, http_span_metadata)
+
+
+def test_create_request_body_take_only_metadata_pymongo_span(
+    function_end_span: dict, pymongo_span: dict, pymongo_span_metadata: dict
+):
+    assert_use_metadata_span_when_needed(function_end_span, pymongo_span, pymongo_span_metadata)
+
+
+def test_create_request_body_take_only_metadata_sql_span(
+    function_end_span: dict, sql_span: dict, sql_span_metadata: dict
+):
+    assert_use_metadata_span_when_needed(function_end_span, sql_span, sql_span_metadata)
+
+
+def test_with_many_spans(function_end_span, http_span, http_span_metadata):
+    expected_result = [function_end_span] + [http_span] * 51 + [http_span_metadata] * 49
+    input_spans = [function_end_span] + [http_span] * 100
+    size = 78080
```

Comment:
ohhh this test will break the next time that someone change something small (e.g. the size HTTP metadata, a new field to HTTP span, etc.)
Can you make it more robust?